### PR TITLE
Piped is here.

### PR DESCRIPTION
**Pull request by lilkidsuave**
NOTE: it is a bring your own config app, like dashy

### CODE REVIEW

File: servapps/Piped/cosmos-compose.json [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                "SmartShield": {
+                   "Enabled": true
+                }
+             }
+          ],
+          "labels": {
+            "cosmos-auto-update": "true"
+          }
+       },
+       "{ServiceName}-backend": {
```
Comment (azukaar) [inline servapps/Piped/cosmos-compose.json:72]: I'm pretty sure this is missing ENV to set the postgres credentials no?

File: servapps/Piped/cosmos-compose.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+          "image": "1337kavin/piped:latest",
+          "restart": "unless-stopped",
+          "volumes": [
+            {
+              "source": "{Context.configPath}",
+              "target": "/app/config.properties",
+              "type": "bind"
```
Comment (azukaar) [inline servapps/Piped/cosmos-compose.json:79]: Config should be volumes (also I think mounting a single file wont work please double check if it does)

File: servapps/Piped/cosmos-compose.json [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+              "source": "{ServiceName}-db",
+              "target": "/var/lib/postgresql/data",
+              "type": "volume"
+              
+            }
+        ],
+          "environment": [
+             "POSTGRES_DB=piped",
+             "POSTGRES_USER=piped",
+             "POSTGRES_PASSWORD=changeme"
```
Comment (azukaar) [inline servapps/Piped/cosmos-compose.json:117]: PLease use the password feature to generate a random password on each install

File: servapps/Piped/cosmos-compose.json [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+       "form": [
+          {
+             "name": "configPath",
+             "label": "What is the path to your config.properties? (See https://github.com/TeamPiped/Piped-Docker/blob/main/template/config.properties and replace anything with _HOSTNAME to match the routes generated)",
+             "initialValue": "{DefaultDataPath}/config/config.properties",
+             "type": "text"
+          }
+       ]
+    },
+    "services": {
```
Comment (azukaar) [inline servapps/Piped/cosmos-compose.json:12]: You need to use the link feature to link the container together if you want them to talk together

https://github.com/azukaar/cosmos-servapps-official/blob/0c7788f67b1b94f39e7fefcb63d444d68b545a95/servapps/Sonarr/cosmos-compose.json#L104

File: servapps/Piped/cosmos-compose.json (hunk 2 above)
Comment (azukaar) [reply servapps/Piped/cosmos-compose.json:79]: (Can this be a folder at all?)

File: servapps/Piped/cosmos-compose.json [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                "Timeout": 14400000,
+                "ThrottlePerMinute": 12000,
+                "BlockCommonBots": true,
+                "SmartShield": {
+                   "Enabled": true
+                }
+             }
+          ],
+          "labels": {
+            "cosmos-auto-update": "true"
```
Comment (azukaar) [inline servapps/Piped/cosmos-compose.json:100]: You should keep the isolate labels too

File: servapps/Piped/cosmos-compose.json [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+          }
+       },
+       "{ServiceName}-postgres": {
+          "image": "postgres:15",
+          "restart": "unless-stopped",
```
Comment (azukaar) [inline servapps/Piped/cosmos-compose.json:105]: DBs should have cosmos-persistent-env labels to keep passwords persistent even when the container is overwritten

**Review by azukaar — CHANGES_REQUESTED** [review]

File: servapps/Piped/cosmos-compose.json (hunk 3 above)
Comment (lilkidsuave) [reply servapps/Piped/cosmos-compose.json:117]: What features

File: servapps/Piped/cosmos-compose.json (hunk 2 above)
Comment (lilkidsuave) [reply servapps/Piped/cosmos-compose.json:79]: Can't be volume, requires outside config file like Dashy.
Can be folder

File: servapps/Piped/cosmos-compose.json (hunk 1 above)
Comment (lilkidsuave) [reply servapps/Piped/cosmos-compose.json:72]: Fixed I think

File: servapps/Piped/cosmos-compose.json (hunk 6 above)
Comment (lilkidsuave) [reply servapps/Piped/cosmos-compose.json:105]: fixed

File: servapps/Piped/cosmos-compose.json (hunk 3 above)
Comment (jwr1) [reply servapps/Piped/cosmos-compose.json]: You can let cosmos generate passwords automatically, like the following: `"POSTGRES_PASSWORD={Passwords.0}"` If you increment the number (`{Passwords.1}`), I'll generate a new password, separate from the first.

**Comment by azukaar** [comment]
Thank you!

**Pull request merged**